android/services/mongodb-remote/src/main/java/com/mongodb/stitch/android/services/mongodb/remote/AsyncChangeStream.java

```diff
@@ -35,49 +35,94 @@
  *                    asynchronously.
  * @param <ChangeEventT> The type of MongoDB change event that this stream internally returns.
  */
-public class AsyncChangeStream<DocumentT, ChangeEventT extends BaseChangeEvent<DocumentT>> extends
-    ChangeStream<Task<ChangeEventT>> {
+public class AsyncChangeStream<DocumentT, ChangeEventT extends BaseChangeEvent<DocumentT>> {
   private final TaskDispatcher dispatcher;
+  private final ChangeStream<ChangeEventT> stream;
 
   /**
    * Initializes a passthrough change stream with the provided underlying event stream.
    *
    * @param stream The event stream.
    * @param dispatcher The event dispatcher.
    */
-  public AsyncChangeStream(final Stream<ChangeEventT> stream,
+  public AsyncChangeStream(final ChangeStream<ChangeEventT> stream,
                            final TaskDispatcher dispatcher) {
-    super(stream);
+    this.stream = stream;
     this.dispatcher = dispatcher;
   }
 
+  /**
+   * Adds a ChangeEventListener to the ChangeStream that will run on every event on the stream.
+   * Multiple ChangeEventListeners can be added to any given stream and they will be removed
+   * when the stream is closed or when the listener is removed.
+   *
+   * @param listener the ChangeEventListener
+   */
+  public void addChangeEventListener(
+      final BaseChangeEventListener<DocumentT, ChangeEventT> listener) {
+    stream.addChangeEventListener(listener);
+  }
+
+  /**
+   * Remove a ChangeEventListener from the ChangeStream.
+   *
+   * @param listener the ChangeEventListener
+   */
+  public void removeChangeEventListener(
```

Comment:
I still think exposing BaseChangeEventListener to the user doesn't feel like the right level of abstraction.

Reply:
How so? I tested to make sure that this is type safe in that you cannot send in the wrong type of listener.